pandas-stubs/core/frame.pyi

```diff
@@ -158,19 +159,16 @@ class _LocIndexerFrame(_LocIndexer):
     ) -> None: ...
 
 class DataFrame(NDFrame, OpsMixin):
-    _ListLike = Union[
-        np.ndarray,
-        list[Dtype],
-        dict[_str, np.ndarray],
-        Sequence,
-        Index,
-        Series,
-    ]
+
     __hash__: ClassVar[None]  # type: ignore[assignment]
 
     def __new__(
         cls,
-        data: _ListLike | DataFrame | dict[Any, Any] | None = ...,
+        data: ListLike
+        | DataFrame
+        | dict[Any, Any]
+        | ItemsView[Hashable, ListLike]
```

Comment:
`DataFrame` calls `np.asarray` or `list`: could probably use `Iterable[ListLike]`?

Reply:
`mypy` doesn't match `dict_items()` with `Iterable[ListLike]`

Reply:
`Iterable[tuple[Hashable, ListLike]]` seems to work: matches the declaration of [ItemView](https://github.com/python/typeshed/blob/3623f923a6647f92d1b276646f69c4927423c449/stdlib/typing.pyi#L536)

Reply:
Thanks.  I think I tried `Iterable[tuple[HashableT, ListLike]]` and that failed, but using `Hashable` instead of `HashableT` works. In next commit